java.lang.module.FindException:  Unable to derive module descriptor for class-model-2.6.1.jar — Invalid module name: 'class' is not a Java identifier

### Context 

I need to build, deploy and run from the command line, Spring Boot Jersey applications that are distributed as Java Platform Module System (JPMS) modules.

### The Problem

Running a Spring Boot Jersey application, with a command like this:

```
$ java --module-path ..:/home/lingocoder/libs/org.glassfish.hk2/class-model/2.6.1/class-model-2.6.1.jar:/home/lingocoder/libs/org.springframework.boot/spring-boot-starter-jersey/2.1.8.RELEASE/spring-boot-starter-jersey-2.1.8.RELEASE.jar:... -m my.module/my.springboot.application.Main
```

...fails with this *`java.lang.module.FindException`*:

```
...
Error occurred during initialization of boot layer
java.lang.module.FindException:  Unable to derive module descriptor for: /home/lingocoder/libs/org.glassfish.hk2/class-model/2.6.1/class-model-2.6.1.jar

Caused by: java.lang.IllegalArgumentException: class.model: Invalid module name: 'class' is not a Java identifier
```
### Steps to Reproduce

1. The same error occurs when running this *`jar`* command:

```
$ jar -f /home/lingocoder/libs/org.glassfish.hk2/class-model/2.6.1/class-model-2.6.1.jar --describe-module
```
2. This is the result of passing that same *`class-model`* jar to *`jdeps`*:

```
$ jdeps --generate-module-info . /home/lingocoder/libs/org.glassfish.hk2/class-model/2.6.1/class-model-2.6.1.jar
Exception in thread "main" java.lang.module.FindException: Unable to derive module descriptor for /home/lingocoder/libs/org.glassfish.hk2/class-model/2.6.1/class-model-2.6.1.jar
        at java.base/jdk.internal.module.ModulePath.readJar(ModulePath.java:646)
        at java.base/jdk.internal.module.ModulePath.readModule(ModulePath.java:329)
        at java.base/jdk.internal.module.ModulePath.scan(ModulePath.java:235)
        at java.base/jdk.internal.module.ModulePath.scanNextEntry(ModulePath.java:188)
        at java.base/jdk.internal.module.ModulePath.findAll(ModulePath.java:164)
        at jdk.jdeps/com.sun.tools.jdeps.ModuleInfoBuilder.<init>(ModuleInfoBuilder.java:85)
        at jdk.jdeps/com.sun.tools.jdeps.JdepsTask$GenModuleInfo.run(JdepsTask.java:943)
        at jdk.jdeps/com.sun.tools.jdeps.JdepsTask.run(JdepsTask.java:574)
        at jdk.jdeps/com.sun.tools.jdeps.JdepsTask.run(JdepsTask.java:533)
        at jdk.jdeps/com.sun.tools.jdeps.Main.main(Main.java:49)
Caused by: java.lang.IllegalArgumentException: class.model: Invalid module name: 'class' is not a Java identifier
        at java.base/jdk.internal.module.Checks.requireModuleName(Checks.java:52)
        at java.base/java.lang.module.ModuleDescriptor$Builder.<init>(ModuleDescriptor.java:1535)
        at java.base/java.lang.module.ModuleDescriptor.newAutomaticModule(ModuleDescriptor.java:2417)
        at java.base/jdk.internal.module.ModulePath.deriveModuleDescriptor(ModulePath.java:506)
        at java.base/jdk.internal.module.ModulePath.readJar(ModulePath.java:642)
        ... 9 more
```

### Related Issues

* [***jna-platform,xalan, batik-script InvalidModuleDescriptorException***](http://bit.ly/jigsawdev1059479)
* [***java.lang.module.FindException under Java 9***](http://bit.ly/RESTEASY1752)
* [***Modular java (eg JDK11) doesn't work correctly with RESTEasy client***](http://bit.ly/RESTEASY1887)
* [***Automatic Module Name***](https://github.com/eclipse-ee4j/glassfish-hk2/issues/458)
* [***Invalid module-info.class in asm-repackaged***](https://github.com/eclipse-ee4j/glassfish-hk2/issues/410)
* [***ArrayIndexOutOfBoundsException at io.github.classgraph.Classfile.getConstantPoolStringOffset***](https://github.com/classgraph/classgraph/issues/319)
* [***java.lang.module.FindException: Automatic-Module-Name: org.xmlunit.xmlunit-core: Invalid module name: 'xmlunit-core' is not a Java identifier***](https://github.com/xmlunit/xmlunit/issues/136)
* [***Upgrade to Tomcat 9.0.26***](https://github.com/spring-projects/spring-boot/issues/18388)

### Environment

```
$ java -version
openjdk version "12" 2019-03-19
OpenJDK Runtime Environment (build 12+33)
OpenJDK 64-Bit Server VM (build 12+33, mixed mode, sharing)
...
OS: Windows amd64
```

Please fix this? TIA.

--- a/class-model/pom.xml
+++ b/class-model/pom.xml
@@ -1,6 +1,7 @@
 <!--
 
     Copyright (c) 2010, 2018 Oracle and/or its affiliates. All rights reserved.
+    Copyright (c) 2019 Payara Services Ltd.
 
     This program and the accompanying materials are made available under the
     terms of the Eclipse Public License v. 2.0, which is available at
@@ -54,6 +55,18 @@
                         </goals>
                     </execution>
                 </executions>
+            </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-jar-plugin</artifactId>
+                <configuration>
+                    <archive>
+                        <manifestFile>${manifest.location}</manifestFile>
+                        <manifestEntries>
+                            <Automatic-Module-Name>org.glassfish.hk2.classmodel</Automatic-Module-Name>
+                        </manifestEntries>
+                    </archive>
+                </configuration>
             </plugin>
         </plugins>
     </build>

--- a/external/aopalliance/pom.xml
+++ b/external/aopalliance/pom.xml
@@ -2,6 +2,7 @@
 <!--
 
     Copyright (c) 2013, 2018 Oracle and/or its affiliates. All rights reserved.
+    Copyright (c) 2019 Payara Services Ltd.
 
     This program and the accompanying materials are made available under the
     terms of the Eclipse Public License v. 2.0, which is available at
@@ -106,6 +107,18 @@
                         </goals>
                     </execution>
                 </executions>
+            </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-jar-plugin</artifactId>
+                <configuration>
+                    <archive>
+                        <manifestFile>${manifest.location}</manifestFile>
+                        <manifestEntries>
+                            <Automatic-Module-Name>org.aopalliance</Automatic-Module-Name>
+                        </manifestEntries>
+                    </archive>
+                </configuration>
             </plugin>
         </plugins>
     </build>

--- a/guice-bridge/pom.xml
+++ b/guice-bridge/pom.xml
@@ -2,6 +2,7 @@
 <!--
 
     Copyright (c) 2013, 2018 Oracle and/or its affiliates. All rights reserved.
+    Copyright (c) 2019 Payara Services Ltd.
 
     This program and the accompanying materials are made available under the
     terms of the Eclipse Public License v. 2.0, which is available at
@@ -47,6 +48,18 @@
                         </goals>
                     </execution>
                 </executions>
+            </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-jar-plugin</artifactId>
+                <configuration>
+                    <archive>
+                        <manifestFile>${manifest.location}</manifestFile>
+                        <manifestEntries>
+                            <Automatic-Module-Name>org.jvnet.hk2.guice.bridge</Automatic-Module-Name>
+                        </manifestEntries>
+                    </archive>
+                </configuration>
             </plugin>
         </plugins>
     </build>

--- a/hk2-api/pom.xml
+++ b/hk2-api/pom.xml
@@ -2,6 +2,7 @@
 <!--
 
     Copyright (c) 2010, 2018 Oracle and/or its affiliates. All rights reserved.
+    Copyright (c) 2019 Payara Services Ltd.
 
     This program and the accompanying materials are made available under the
     terms of the Eclipse Public License v. 2.0, which is available at
@@ -56,6 +57,18 @@
                     <!-- -Djava.security.debug=access,failure,domain -->
                 </configuration>
              </plugin>
+             <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-jar-plugin</artifactId>
+                <configuration>
+                    <archive>
+                        <manifestFile>${manifest.location}</manifestFile>
+                        <manifestEntries>
+                            <Automatic-Module-Name>org.glassfish.hk2.api</Automatic-Module-Name>
+                        </manifestEntries>
+                    </archive>
+                </configuration>
+            </plugin>
         </plugins>
     </build>
 

--- a/hk2-configuration/hk2-integration/pom.xml
+++ b/hk2-configuration/hk2-integration/pom.xml
@@ -2,6 +2,7 @@
 <!--
 
     Copyright (c) 2014, 2018 Oracle and/or its affiliates. All rights reserved.
+    Copyright (c) 2019 Payara Services Ltd.
 
     This program and the accompanying materials are made available under the
     terms of the Eclipse Public License v. 2.0, which is available at
@@ -58,6 +59,18 @@
                         </goals>
                     </execution>
                 </executions>
+            </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-jar-plugin</artifactId>
+                <configuration>
+                    <archive>
+                        <manifestFile>${manifest.location}</manifestFile>
+                        <manifestEntries>
+                            <Automatic-Module-Name>org.glassfish.hk2.configuration</Automatic-Module-Name>
+                        </manifestEntries>
+                    </archive>
+                </configuration>
             </plugin>
         </plugins>
     </build>

--- a/hk2-configuration/manager/pom.xml
+++ b/hk2-configuration/manager/pom.xml
@@ -2,6 +2,7 @@
 <!--
 
     Copyright (c) 2014, 2018 Oracle and/or its affiliates. All rights reserved.
+    Copyright (c) 2019 Payara Services Ltd.
 
     This program and the accompanying materials are made available under the
     terms of the Eclipse Public License v. 2.0, which is available at
@@ -58,6 +59,18 @@
                         </goals>
                     </execution>
                 </executions>
+            </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-jar-plugin</artifactId>
+                <configuration>
+                    <archive>
+                        <manifestFile>${manifest.location}</manifestFile>
+                        <manifestEntries>
+                            <Automatic-Module-Name>org.glassfish.hk2.configuration.hub</Automatic-Module-Name>
+                        </manifestEntries>
+                    </archive>
+                </configuration>
             </plugin>
         </plugins>
     </build>

--- a/hk2-core/pom.xml
+++ b/hk2-core/pom.xml
@@ -2,6 +2,7 @@
 <!--
 
     Copyright (c) 2010, 2018 Oracle and/or its affiliates. All rights reserved.
+    Copyright (c) 2019 Payara Services Ltd.
 
     This program and the accompanying materials are made available under the
     terms of the Eclipse Public License v. 2.0, which is available at
@@ -49,6 +50,18 @@
                         </goals>
                     </execution>
                 </executions>
+            </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-jar-plugin</artifactId>
+                <configuration>
+                    <archive>
+                        <manifestFile>${manifest.location}</manifestFile>
+                        <manifestEntries>
+                            <Automatic-Module-Name>org.glassfish.hk2.core</Automatic-Module-Name>
+                        </manifestEntries>
+                    </archive>
+                </configuration>
             </plugin>
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>

--- a/hk2-extras/pom.xml
+++ b/hk2-extras/pom.xml
@@ -2,6 +2,7 @@
 <!--
 
     Copyright (c) 2014, 2018 Oracle and/or its affiliates. All rights reserved.
+    Copyright (c) 2019 Payara Services Ltd.
 
     This program and the accompanying materials are made available under the
     terms of the Eclipse Public License v. 2.0, which is available at
@@ -47,6 +48,18 @@
                         </goals>
                     </execution>
                 </executions>
+            </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-jar-plugin</artifactId>
+                <configuration>
+                    <archive>
+                        <manifestFile>${manifest.location}</manifestFile>
+                        <manifestEntries>
+                            <Automatic-Module-Name>org.glassfish.hk2.extras</Automatic-Module-Name>
+                        </manifestEntries>
+                    </archive>
+                </configuration>
             </plugin>
         </plugins>
     </build>

--- a/hk2-jmx/pom.xml
+++ b/hk2-jmx/pom.xml
@@ -2,6 +2,7 @@
 <!--
 
     Copyright (c) 2010, 2018 Oracle and/or its affiliates. All rights reserved.
+    Copyright (c) 2019 Payara Services Ltd.
 
     This program and the accompanying materials are made available under the
     terms of the Eclipse Public License v. 2.0, which is available at
@@ -47,6 +48,18 @@
                         </goals>
                     </execution>
                 </executions>
+            </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-jar-plugin</artifactId>
+                <configuration>
+                    <archive>
+                        <manifestFile>${manifest.location}</manifestFile>
+                        <manifestEntries>
+                            <Automatic-Module-Name>org.glassfish.hk2.jmx</Automatic-Module-Name>
+                        </manifestEntries>
+                    </archive>
+                </configuration>
             </plugin>
         </plugins>
     </build>

--- a/hk2-locator/pom.xml
+++ b/hk2-locator/pom.xml
@@ -2,6 +2,7 @@
 <!--
 
     Copyright (c) 2010, 2018 Oracle and/or its affiliates. All rights reserved.
+    Copyright (c) 2019 Payara Services Ltd.
 
     This program and the accompanying materials are made available under the
     terms of the Eclipse Public License v. 2.0, which is available at
@@ -59,7 +60,18 @@
                     <!-- -Djava.security.debug=access,failure,domain -->
                 </configuration>
             </plugin>
-
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-jar-plugin</artifactId>
+                <configuration>
+                    <archive>
+                        <manifestFile>${manifest.location}</manifestFile>
+                        <manifestEntries>
+                            <Automatic-Module-Name>org.glassfish.hk2.locator</Automatic-Module-Name>
+                        </manifestEntries>
+                    </archive>
+                </configuration>
+            </plugin>
         </plugins>
     </build>
     <dependencies>

--- a/hk2-runlevel/pom.xml
+++ b/hk2-runlevel/pom.xml
@@ -2,6 +2,7 @@
 <!--
 
     Copyright (c) 2010, 2018 Oracle and/or its affiliates. All rights reserved.
+    Copyright (c) 2019 Payara Services Ltd.
 
     This program and the accompanying materials are made available under the
     terms of the Eclipse Public License v. 2.0, which is available at
@@ -65,6 +66,18 @@
                 <configuration>
                     <!-- Turns on debug logging -->
                     <argLine>-Dorg.jvnet.hk2.logger.debugToStdout=false -Dorg.jvnet.hk2.properties.debug.runlevel.context=false ${surefireArgLineExtra}</argLine>
+                </configuration>
+            </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-jar-plugin</artifactId>
+                <configuration>
+                    <archive>
+                        <manifestFile>${manifest.location}</manifestFile>
+                        <manifestEntries>
+                            <Automatic-Module-Name>org.glassfish.hk2.runlevel</Automatic-Module-Name>
+                        </manifestEntries>
+                    </archive>
                 </configuration>
             </plugin>
         </plugins>

--- a/hk2-utils/pom.xml
+++ b/hk2-utils/pom.xml
@@ -2,6 +2,7 @@
 <!--
 
     Copyright (c) 2010, 2018 Oracle and/or its affiliates. All rights reserved.
+    Copyright (c) 2019 Payara Services Ltd.
 
     This program and the accompanying materials are made available under the
     terms of the Eclipse Public License v. 2.0, which is available at
@@ -42,6 +43,7 @@
                     <archive>
                         <manifestFile>target/classes/META-INF/MANIFEST.MF</manifestFile>
                         <manifestEntries>
+                            <Automatic-Module-Name>org.glassfish.hk2.utilities</Automatic-Module-Name>
                             <service>foo</service>
                         </manifestEntries>
                     </archive>

--- a/maven-plugins/consolidatedbundle-maven-plugin/pom.xml
+++ b/maven-plugins/consolidatedbundle-maven-plugin/pom.xml
@@ -2,6 +2,7 @@
 <!--
 
     Copyright (c) 2010, 2018 Oracle and/or its affiliates. All rights reserved.
+    Copyright (c) 2019 Payara Services Ltd.
 
     This program and the accompanying materials are made available under the
     terms of the Eclipse Public License v. 2.0, which is available at
@@ -53,6 +54,18 @@
                 <configuration>
                     <source>1.6</source>
                     <target>1.6</target>
+                </configuration>
+            </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-jar-plugin</artifactId>
+                <configuration>
+                    <archive>
+                        <manifestFile>${manifest.location}</manifestFile>
+                        <manifestEntries>
+                            <Automatic-Module-Name>org.glassfish.hk2.maven.bundle</Automatic-Module-Name>
+                        </manifestEntries>
+                    </archive>
                 </configuration>
             </plugin>
         </plugins>

--- a/maven-plugins/hk2-inhabitant-generator/pom.xml
+++ b/maven-plugins/hk2-inhabitant-generator/pom.xml
@@ -2,6 +2,7 @@
 <!--
 
     Copyright (c) 2010, 2018 Oracle and/or its affiliates. All rights reserved.
+    Copyright (c) 2019 Payara Services Ltd.
 
     This program and the accompanying materials are made available under the
     terms of the Eclipse Public License v. 2.0, which is available at
@@ -91,6 +92,11 @@
                          </goals>
                          <phase>process-test-classes</phase>
                          <configuration>
+                             <archive>
+                                 <manifestEntries>
+                                     <Automatic-Module-Name>org.glassfish.hk2.maven.generator</Automatic-Module-Name>
+                                 </manifestEntries>
+                             </archive>
                              <excludes>
                                  <exclude>**/negative/*</exclude>
                              </excludes>

--- a/maven-plugins/osgiversion-maven-plugin/pom.xml
+++ b/maven-plugins/osgiversion-maven-plugin/pom.xml
@@ -1,6 +1,7 @@
 <!--
 
     Copyright (c) 2010, 2018 Oracle and/or its affiliates. All rights reserved.
+    Copyright (c) 2019 Payara Services Ltd.
 
     This program and the accompanying materials are made available under the
     terms of the Eclipse Public License v. 2.0, which is available at
@@ -49,5 +50,22 @@
             <artifactId>maven-osgi</artifactId>
         </dependency>
     </dependencies>
+    
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-jar-plugin</artifactId>
+                <configuration>
+                    <archive>
+                        <manifestFile>${manifest.location}</manifestFile>
+                        <manifestEntries>
+                            <Automatic-Module-Name>org.glassfish.hk2.maven.osgiversion</Automatic-Module-Name>
+                        </manifestEntries>
+                    </archive>
+                </configuration>
+            </plugin>
+        </plugins>
+    </build>
     
 </project>

--- a/spring-bridge/pom.xml
+++ b/spring-bridge/pom.xml
@@ -2,6 +2,7 @@
 <!--
 
     Copyright (c) 2013, 2018 Oracle and/or its affiliates. All rights reserved.
+    Copyright (c) 2019 Payara Services Ltd.
 
     This program and the accompanying materials are made available under the
     terms of the Eclipse Public License v. 2.0, which is available at
@@ -47,6 +48,18 @@
                         </goals>
                     </execution>
                 </executions>
+            </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-jar-plugin</artifactId>
+                <configuration>
+                    <archive>
+                        <manifestFile>${manifest.location}</manifestFile>
+                        <manifestEntries>
+                            <Automatic-Module-Name>org.glassfish.hk2.spring.bridge</Automatic-Module-Name>
+                        </manifestEntries>
+                    </archive>
+                </configuration>
             </plugin>
         </plugins>
     </build>

